### PR TITLE
fix(es/minifier): remove recklessly inline

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

Fix an issue found in #5526 

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_ecma_minifier/tests/fixture/issues/issue-2604/1/output.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-                return eval(a);
-            } catch (b) {}
-        }
-        function Rb(a1) {
-            return p() ? JSON : za();
-        }
-        h.get = function(a1) {
-            return null == a1 ? null : Rb(a1);
-        };
+        !function() {
```
Comment (Austaras) [inline crates/swc_ecma_minifier/tests/fixture/issues/issue-2604/1/output.js:3]: Don't know why but eval prevents inlining

File: crates/swc_ecma_minifier/tests/projects/output/angular-1.2.5.js [hunk 2]
```diff
@@ -913,9 +913,11 @@
         var pollTimeout, pollFns = [];
         self.addPollFn = function(fn) {
             var setTimeout2;
-            return isUndefined(pollTimeout) && (setTimeout2 = setTimeout1, forEach(pollFns, function(pollFn) {
-                pollFn();
-            }), pollTimeout = setTimeout2(check, 100)), pollFns.push(fn), fn;
```
Comment (Austaras) [inline crates/swc_ecma_minifier/tests/projects/output/angular-1.2.5.js:918]: This is a bug -- check disappers

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_minifier

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

**Review by kdy1 — APPROVED** [review]
Thanks!

File: crates/swc_ecma_minifier/tests/fixture/issues/issue-2604/1/output.js (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/issues/issue-2604/1/output.js:3]: It was intentional as we should preserve scope around `eval`

**Pull request merged**